src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -731,6 +731,7 @@ protected bool TakeStartLine(SocketInput input)
                     // URI was encoded, unescape and then parse as utf8
                     pathEnd = UrlPathDecoder.Unescape(pathBegin, pathEnd);
                     requestUrlPath = pathBegin.GetUtf8String(pathEnd);
+                    requestUrlPath = PathNormalizer.NormalizeToNFC(requestUrlPath);
```

Comment:
@benaadams It's not possible to remove dot segments without incurring extra allocations OR making significant changes to memory blocks.

There are two ways to do it (that I thought of, if you have something better in mind please share :)):

1) Move forward in the string and keep a stack of segments to figure out which ones to skip. The problem here is that we'd have to allocate a stack and incur whatever cost `Push()` and `Pop()` would impose.

2) Go backward in the string and skip segments as `..`s are found. This is what I am doing now.

We can't do 2 with blocks/iterators at this time because that would require being able to navigate back in blocks. While adding a `back` pointer to blocks isn't a super hard change, I wouldn't like to introduce that change at this time.